pkg/util/pki/csr.go

```diff
@@ -238,15 +275,11 @@ func GenerateTemplateFromCertificateRequest(cr *v1alpha1.CertificateRequest) (*x
 		NotBefore:             time.Now(),
 		NotAfter:              time.Now().Add(certDuration),
 		// see http://golang.org/pkg/crypto/x509/#KeyUsage
-		KeyUsage:    keyUsage(cr.Spec.IsCA),
+		KeyUsage:    keyUsages,
+		ExtKeyUsage: extKeyUsages,
 		DNSNames:    csr.DNSNames,
 		IPAddresses: csr.IPAddresses,
 		URIs:        csr.URIs,
-		// TODO: we should expose ExtKeyUsage via the API and not set x509.ExtKeyUsageClientAuth
-		// by default. This is a known change in behaviour between the Certificate and CertificateRequest
-		// controller and should be rectified before the CertificateRequest feature exits
-		// alpha.
-		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
```

Comment:
should I not be adding these capabilities in line 265? Didn't want to break backwards compatibility